### PR TITLE
Add new methods in update_class()

### PR DESCRIPTION
**Pull request by oscar6echo**
Reference #11281.

Currently the autoreload magic only upgrades existing methods but ignores new ones.  
This PR enables to captures new methods - very useful in incremental dev.

### CODE REVIEW

**Comment by Carreau** [comment]
let's see if we can find how to fold your test case into the test suite.

**Comment by Carreau** [comment]
I've tried adding some tests. 
Let me know if that's what you were looking for.

I haven't asserted the values of c.a c.b c.toto and I am jetlagged, so may have done stupid things.

File: IPython/testing/iptest.py [hunk 1]
```diff
@@ -385,6 +385,12 @@ def run_iptest():
     if '--with-xunit' in sys.argv and not hasattr(Xunit, 'orig_addError'):
         monkeypatch_xunit()
 
+    arg1 = sys.argv[1]
+    if arg1.startswith('IPython/'):
+        if arg1.endswith('.py'):
+            arg1 = arg1[:-3]
+        sys.argv[1] = arg1.replace('/', '.')
```
Comment (Carreau) [inline IPython/testing/iptest.py]: That allows to run iptest with a filepath which is a little more confenient for bash autocompletion.

**Comment by oscar6echo** [comment]
The tests look fine to me. Thank you, I would not have been able to write them.  
If you think the whatsnew/pr/improve-autoreload.md is too verbose, feel free to cull.

**Comment by Carreau** [comment]
> The tests look fine to me. Thank you, I would not have been able to write them.

It's ok they were not easy to write, testing this in one of the tricky bits. 


> If you think the whatsnew/pr/improve-autoreload.md is too verbose, feel free to cull.

all the `watsnew/pr/*` will be merged in the changelog at release time, and well be edited then.

**Pull request merged**